src/Cardano/Environment.hs

```diff
@@ -0,0 +1,168 @@
+-- @
+instance Exception ErrMissingOrInvalidEnvVar where
+    displayException (ErrMissingOrInvalidEnvVar n cmd ctx) = pretty $ mempty
+        <> nameF "Missing or invalid ENV var"
+            ( "\n  ENV[" <> build n <> "] = " <> ctxF )
+      where
+        ctxF = case ctx of
+            Nothing -> "?"
+                <> "\n\nWhat about trying to provide a valid ENV var "
+                <> "`" <> build n <> "=value " <> build cmd <> "` ?"
+            Just (v, err) ->
+                let
+                    pad = length n + (length v `div` 2) + 11
+                in
+                    build v
+                        <> "\n  " <> padLeftF @Text pad ' ' "|    "
+                        <> "\n  " <> padLeftF @Text pad ' ' "|    "
+                        <> "\n  " <> padLeftF @Text pad ' ' "*--> "
+                        <> build err
+
+-- | Lookup the environment for a given variable
+unsafeLookupEnv
+    :: FromText a
+    => String
+    -> a
+unsafeLookupEnv k = unsafePerformIO $ do
+    cmd <- getProgName
+    v <- lookupEnv k >>= \case
+        Just v -> return v
+        Nothing -> throwIO $ ErrMissingOrInvalidEnvVar
+            { name = k
+            , command =  cmd
+            , additionalContext = Nothing
+            }
+    case fromText (T.pack v) of
+        Right a -> return a
+        Left err -> throwIO $ ErrMissingOrInvalidEnvVar
+            { name = k
+            , command = cmd
+            , additionalContext = Just (v, err)
+            }
+
+{-------------------------------------------------------------------------------
+                                 Environment
+-------------------------------------------------------------------------------}
+
+-- | Available network options. 'Local' means a local cluster running on the
+-- host machine.
+data Network = Mainnet | Testnet | Staging | Local
+    deriving (Generic, Show, Eq, Enum)
+
+instance FromText Network where
+    fromText = \case
+        "mainnet" -> Right Mainnet
+        "testnet" -> Right Testnet
+        "staging" -> Right Staging
+        "local" -> Right Local
+        s -> Left $ TextDecodingError $ T.unpack s
+            <> " is neither \"mainnet\", \"testnet\", \"staging\" nor \"local\"."
+
+instance ToText Network where
+    toText = \case
+        Mainnet -> "mainnet"
+        Testnet -> "testnet"
+        Staging -> "staging"
+        Local -> "local"
+
+-- | Get the current target 'Network' from the Environment.
+--
+-- Throws a runtime exception is the ENV var isn't set or, is invalid.
+network :: Network
+network =
+    unsafeLookupEnv "NETWORK"
+{-# NOINLINE network #-}
+
+newtype ProtocolMagic = ProtocolMagic Int32
+    deriving (Generic, Show)
+
+-- | Get the 'ProtocolMagic' corresponding to a given 'Network'.
+--
+-- Note that the 'ProtocolMagic' for 'Local' and 'Testnet' are the same.
+protocolMagic :: Network -> ProtocolMagic
+protocolMagic = \case
+    Mainnet -> ProtocolMagic 764824073
```

Comment:
I hope these hardcoded values are right - because I am not sure where to look up for the proof/source that these are the right numbers?

Maybe it would be usefull for other readers to add a comment where from or why these specific numbers were picked

Reply:
Actually, I am not even sure where I got them from :| ... I think I now mostly go for the ~/.hermes configuration files generated by the http bridge to be honest...